packages/run-protocol/test/vaultFactory/vault-contract-wrapper.js

```diff
@@ -21,7 +21,13 @@ const BASIS_POINTS = 10000n;
 const SECONDS_PER_HOUR = 60n * 60n;
 const DAY = SECONDS_PER_HOUR * 24n;
 
-/** @type {ContractStartFn} */
+/**
+ * @type {ContractStartFn<
+ * undefined,
+ * { makeAdjustBalancesInvitation: unknown, mintRun: unknown },
+ * {},
+ * {feeMintAccess: FeeMintAccess}>}
+ */
```

Comment:
It is and I think we can do better, but I'm going to save that for another PR. The problem I see is that the contract's `start` function is kind of like a constructor for a contract instance and what is returns is the definition of what it should return. IOW, we should rely on the type of what it actually returns instead of declaring a type and checking that it returned that.

We do want some constraints on what it can return, which I think we can accomplish with a utility type. See this [demo validating a CopyRecord](https://www.typescriptlang.org/play?ssl=1&ssc=1&pln=2&pc=1#code/C4TwDgpgBACghgZwXARgG2gXigOwK5ppQA+ueAtihAE4lQLDUCWOA5gNxQD0X9weKALAAoEaEhQAwgHswIAEoQAxtOoATADwAVTIpXqNDZmwA0sRMnQQAfNajYA3iKhRqEOGuk40IKAG0YKBYoAGsIEGkAMygtAF0ALigACi0A2KgIAA9gCBw1BHMkVAwoAH4YtKhEnAgANxoASnYRAF9m0WEVHAZXeygnYRc4RIBySOlpEZNnKBREgCZpwaglRIAROByltpERagA6OH2upU2kpr39lH3gaQAxJkyINXP2g6V9gFUtSXaRHl6pxwOGkwFm0EY7hyaigiFhUlkCmUqjUIi6PSU1ESMjkehRGnEECirjs2Gof06ByOJzOF0pVxu90ez1e-14mOOQQKhKgIxq9WoIzR7y+PwpAIA7gALXJQcgQSg0ApwNw4EZgkIgiU4NFeHp4THYxF49R9fCEWHKnAgdoGqkINSRTgA+BFKwiIA). We could have a `Contract` type that has fields for each of the relevant types, and create that from the `start` function. PR forthcoming.

Reply:
Contract type work: https://github.com/Agoric/agoric-sdk/issues/4845

Reply:
> IOW, we should rely on the type of what it actually returns instead of declaring a type and checking that it returned that.

strongly agree.